<launch>

  <arg name="marker_size" default="0.05" />

  <arg name="video_device" default="/dev/video0" />

  <arg name="camera_info_file" default="file://$(find usb_cam)/config/usb_cam.yaml" />


  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
    <param name="video_device" value="$(arg video_device)" />
    <param name="pixel_format" value="yuyv" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="framerate" value="30" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap" />

    <param name="camera_info_url" value="$(arg camera_info_file)" />
  </node>


  <node name="aruco_detector" pkg="beginner_tutorials" type="aruco_detector.py" output="screen">
    <param name="marker_length" value="$(arg marker_size)" />

    <remap from="/usb_cam/image_raw" to="/usb_cam/image_raw" />
  </node>


  <node name="robot_pose" pkg="beginner_tutorials" type="robot_pose.py" output="screen">
    </node>


  <node name="image_view" pkg="rqt_image_view" type="rqt_image_view" args="/aruco_detector/debug_image" />

</launch>